ToDoWebApi/Controller/UserController.cs

```diff
@@ -61,10 +65,10 @@ public IActionResult AddUser([FromBody] CreateUserViewModel newUser)
     }
 
     [HttpPut("{id}")]
-    public IActionResult UpdateUser([FromBody] UpdateUserModel updateUser, int id)
+    public IActionResult UpdateUser([FromBody] UpdateUserViewModel updateUser, int id)
     {
         UpdateUserCommand command = new UpdateUserCommand(_context, id);
-        command.Id = id;
+        command.Model.Id = id;
         command.Model = updateUser;
```

Comment:
Burada yaptığın hata bu command'i test etmediğini gösteriyor. command'in modeli yokken model'in Id sine değer atamanı yukarıda yazmıştım. Burada yaptığın hata yapılacak bir hata değil, gidip error var diye aynı satırda düzeltmişsin ama güncelleme yaparken daha dikkatli olman, sonrasında da güncellediğin yerleri gerçekten test etmen lazım.

command'in daha olmayan, null olan modelini önce constructor'a id vererek güncellemeye çalışıyorsun, sonra olmayan model'in id'sini manuel güncellemeye çalışıyorsun, sonra da güncellemeye çalıştığın değerin üzerine yazıyorsun. Böyle bir programlama şekli, bu metodun da çalışma ihtimali yok. Test etmeden yapılmış olmaz.

Reply:
Düzeltildi